armi/reactor/tests/test_blocks.py

```diff
@@ -1522,13 +1509,196 @@ def test_getPinCoords(self):
+        # the inner dimensions is defined by whatever thickness is necessary to have
+        # the desired area fraction.
+        # The second way is shown in the second half of this test.
+        cartBlock = blocks.CartesianBlock("TestCartBlock")
+
+        hexComponentArea = areaFractions[0] * rectTotalArea
+
+        # Picking 1st material to use for the hex component here, but really the choice
+        # is arbitrary.
+        # area grows quadratically with outer dimensions.
+        # Note there are infinitely many inner dims that would preserve area,
+        # this is just one of them.
+        innerDims = [dim * areaFractions[0] ** 0.5 for dim in desiredPitch]
+        self.assertAlmostEqual(
+            hexComponentArea, rectTotalArea - innerDims[0] * innerDims[1]
+        )
+
+        rectArgs = {
+            "lengthOuter": desiredPitch[0],
+            "lengthInner": innerDims[0],
+            "widthOuter": desiredPitch[1],
+            "widthInner": innerDims[1],
+            "mult": 1.0,
+        }
+        rectArgs.update(compArgs)
+        pitchDefiningComponent = components.Rectangle(
+            "pitchComp", materials[0], **rectArgs
+        )
+        cartBlock.addComponent(pitchDefiningComponent)
+
+        # Rectangle component is added, now add the rest as unshaped.
+        for aFrac, material in zip(areaFractions[1:], materials[1:]):
+            unshapedArgs = {"area": rectTotalArea * aFrac}
+            unshapedArgs.update(compArgs)
+            name = f"unshaped {material}"
+            comp = components.UnshapedComponent(name, material, **unshapedArgs)
+            cartBlock.addComponent(comp)
+
+        self.assertEqual(desiredPitch, cartBlock.getPitch())
+        self.assertAlmostEqual(rectTotalArea, cartBlock.getMaxArea())
+        self.assertAlmostEqual(sum(c.getArea() for c in cartBlock), rectTotalArea)
+
+        # For this second way, we will simply define the 3 components as unshaped, with
+        # the desired area fractions, and make a 4th component that is an infinitely
+        # thin rectangle with desired pitch. the downside of this method is that now
+        # the block has a fourth component with no volume.
+        cartBlock = blocks.CartesianBlock("TestCartBlock")
+        for aFrac, material in zip(areaFractions, materials):
+            unshapedArgs = {"area": rectTotalArea * aFrac}
+            unshapedArgs.update(compArgs)
+            name = f"unshaped {material}"
+            comp = components.UnshapedComponent(name, material, **unshapedArgs)
+            cartBlock.addComponent(comp)
+
+        # We haven't set a pitch defining component this time so set it now with 0 area.
+        pitchDefiningComponent = components.Rectangle(
+            "pitchComp",
+            "Void",
+            lengthOuter=desiredPitch[0],
+            lengthInner=desiredPitch[0],
+            widthOuter=desiredPitch[1],
+            widthInner=desiredPitch[1],
+            mult=1,
```

Comment:
I recommend just leaving this 2nd way test out to keep the testmaintenance burden low. You demonstrated this approach fine in the hex test.